addons/website_event_crm/models/event_lead_rule.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models
+
+
+class EventLeadRule(models.Model):
+
+    _inherit = "event.lead.rule"
+
+    def _run_on_registrations(self, registrations):
+        leads = super()._run_on_registrations(registrations)
+        if len(self.env['res.lang'].get_installed()) < 2:
+            return leads
+
+        for lead in leads:
+            # When a lead has several registrations, they should all relate to the same visitor
+            visitors = lead.registration_ids.visitor_id
+            if visitors:
+                lead.visitor_ids = visitors
+                if not lead.lang_id and visitors.lang_id and visitors[0].website_id.language_count > 1:
```

Comment:
Can you explain: `and visitors[0].website_id.language_count > 1` ? What is the use?
I also notice we could have a different visitor between the one giving the lang_id to lead and visitors[0].

Like Visitor 0 has no lang_id but visitor1 has one. -> condition on website of first visitor (I guess it should not happen)
Can we assume they all have the same website_id? then we could just pick first visitor having a lang and check their website with something like:

`visitor = next((visitor for visitor in visitors if visitor.lang_id), False)` and then use visitor in various conditions

Reply:
Seems like it would be okay yes

These are just "precautions" in case there are several visitors but really, with the Lead generation rule, 1 lead = 1 visitor from what I've tested

`and visitors[0].website_id.language_count > 1` this is because imo, if the website only has one language, then it doesn't really make sense to assign it to a lead as any visitor would get that language